kernel/work.c

```diff
@@ -646,6 +646,8 @@ static void work_queue_main(void *workq_ptr, void *p2, void *p3)
 			 * submissions.
 			 */
 			(void)z_sched_wake_all(&queue->drainq, 1, NULL);
+		} else {
+			__ASSERT(false, "Unknown condition in work queue.");
```

Comment:
this is breaking tests, so obviously we there is a case where the above conditions are not met which seems to be correct.

Reply:
Thank you! I will fix this.